src/JsonSchema/Constraints/TypeConstraint.php

```diff
@@ -185,6 +185,10 @@ protected function validateType(&$value, $type, $coerce = false)
         }
 
         if ('array' === $type) {
+            if ($coerce) {
+                $value = $this->toArray($value);
```

Comment:
Good suggestion - let's do that then. And if someone asks for it, then we can add it at that point.

I don't supposed you know whether @epoberezkin has stated his rationale for doing it that way in ajv?